ProjNet/CoordinateSystems/Projections/MapProjection.cs

```diff
@@ -264,79 +318,279 @@ public override string XML
+        /// Abstract method to convert a point (lon, lat, z) in radians to (x, y, z) in meters
+        /// </summary>
+        /// <param name="lon">The longitude in radians</param>
+        /// <param name="lat">The latitude in radians</param>
+        /// <param name="z">The z-ordinate</param>
+        /// <returns>Converted point.</returns>
+        protected abstract (double x, double y, double z) RadiansToMeters(double lon, double lat, double z);
+
+        /// <summary>
+        /// Converts a point (lon, lat, z) in degrees to (x, y, z) in meters
+        /// </summary>
+        /// <param name="lon">The longitude in degree</param>
+        /// <param name="lat">The latitude in degree</param>
+        /// <param name="z">The z-ordinate</param>
+        /// <returns>Converted point.</returns>
+        protected (double x, double y, double z) DegreesToMeters(double lon, double lat, double z)
+        {
+            return RadiansToMeters(Degrees2Radians(lon), Degrees2Radians(lat), z);
+        }
+
+        /// <summary>
+        /// Converts a point from degrees to target units
+        /// </summary>
+        /// <param name="lon">The longitude in degree</param>
+        /// <param name="lat">The latitude in degree</param>
+        /// <param name="z">The z-ordinate</param>
+        /// <returns>Converted point.</returns>
+        protected virtual (double x, double y, double z) DegreesToTarget(double lon, double lat, double z)
+        {
+            //x = Degrees2Radians(x);
+            //y = Degrees2Radians(y);
+            double x, y;
+            (x, y, z) = DegreesToMeters(lon, lat, z);
+            (x, y) = MetersToTarget(x, y);
+            return (x, y, z);
+        }
+
         /// <summary>
-        /// Function to transform from meters to degrees
+        /// Converts a series of points from degrees to target units
         /// </summary>
-        /// <param name="p">The ordinates of the point</param>
-        /// <returns>The transformed ordinates</returns>
-        public double[] MetersToDegrees(double[] p)
+        /// <param name="inLons">A series of longitudes in degree</param>
+        /// <param name="inLats">A series of latitudes in degree</param>
+        /// <param name="inZs">A series of z-ordinate values</param>
+        /// <param name="outXs">A buffer for x-ordinate values</param>
+        /// <param name="outYs">A buffer for y-ordinate values</param>
+        /// <param name="outZs">A buffer for z-ordinate values</param>
+        /// <param name="strideX">A stride value for x-ordinates</param>
+        /// <param name="strideY">A stride value for y-ordinates</param>
+        /// <param name="strideZ">A stride value for z-ordinates</param>
+        protected virtual void DegreesToTarget(ReadOnlySpan<double> inLons, ReadOnlySpan<double> inLats, ReadOnlySpan<double> inZs, 
+            Span<double> outXs, Span<double> outYs, Span<double> outZs,
+            int strideX = 1, int strideY = 1, int strideZ = 1)

```

Comment:
see comments in `MathTransform` about default parameters and separate strides for inputs / outputs